### PR TITLE
$: Use rz_flag_get to check if flag exists

### PR DESCRIPTION
**Pull request by kazarmy**
**Your checklist for this pull request**
- [X] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [X] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [X] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

This pr makes `cmd_alias` use `rz_flag_get` instead of `rz_num_get` to check whether a flag exists, so that the test works as expected.

**Test plan**

All builds are green.

**Closing issues**

...

### CODE REVIEW

**Review by XVilka — APPROVED** [review]

File: librz/core/cmd/cmd_alias.c [hunk 1]
```diff
@@ -117,9 +117,9 @@ RZ_IPI RzCmdStatus rz_alias_handler(RzCore *core, int argc, const char **argv) {
 				rz_core_cmd0(core, v);
 			}
 		} else {
-			ut64 at = rz_num_get(core->num, buf + 1);
-			if (at != UT64_MAX) {
-				rz_core_seek(core, at, true);
+			RzFlagItem *flag = rz_flag_get(core->flags, buf + 1);
+			if (flag) {
+				rz_core_seek(core, flag->offset, true);
```
Comment (ret2libc) [inline librz/core/cmd/cmd_alias.c:122]: what is the purpose of this whole piece of code?

Comment (kazarmy) [reply librz/core/cmd/cmd_alias.c:122]: $-flag
(from `$??`)

File: librz/core/cmd/cmd_alias.c [hunk 2]
```diff
@@ -117,9 +117,9 @@ RZ_IPI RzCmdStatus rz_alias_handler(RzCore *core, int argc, const char **argv) {
 				rz_core_cmd0(core, v);
 			}
 		} else {
-			ut64 at = rz_num_get(core->num, buf + 1);
```
Comment (wargio) [inline librz/core/cmd/cmd_alias.c:120]: doesn't removing this forbid doing `flag.something+0x123` ?

Comment (kazarmy) [reply librz/core/cmd/cmd_alias.c:120]: no, that should still work fine

flag+off

it does forbid you from doing `$0x123` to mean `s 0x123` but the latter definitely seems better

**Review by wargio — APPROVED** [review]

File: librz/core/cmd/cmd_alias.c (hunk 2 above)
Comment (kazarmy) [reply librz/core/cmd/cmd_alias.c:120]: I don't know whether there's a use case for allowing `$foo+1` to mean `s foo+1`, unless one wants everything to be $-prefixed for the bling ... also not sure whether it's wise for `$` to be overloaded as shown below:

$-flag
$-alias

but that's not my concern right now.

Comment (wargio) [reply librz/core/cmd/cmd_alias.c:120]: no, i think is ok.

**Pull request merged**